docs/write-custom-pipeline.md

```diff
@@ -0,0 +1,163 @@
+# Writing a new training pipeline
+
+This tutorial will walk you through how to set up your first custom training pipeline.
+The goal of this example pipeline is to orchestrate a classifier that can distinguish
+between daisies and roses.
+
+In this tutorial we will 
+- learn which configuration yaml files we have to write to get our pipeline running
+- have a look how the dataset, training, prediction, etc. configurations are connected with each other 
+- train a binary classifier with niceML
+
+## Prerequisites
+Before we get started, please ensure that you have followed the [Getting Started](getting-started.md)
+tutorial and initialized a new niceML project.
+
+## Step 1: Data ingestion
+Download the images from [Kaggle](https://www.kaggle.com/datasets/alxmamaev/flowers-recognition)
+and store the roses and daisies images in a `data` folder in your niceML project.
+
+## Step 2: Configure the pipeline job
+Let us start configuring at the highest level of niceML: The job 
+configuration. For this, navigate to `configs/jobs/job_train/job_train_cls` 
+and create a new file `job_train_cls_binary_flowers.yaml`.
+
+Paste the content of `configs/jobs/job_train/job_train_cls/job_train_cls_binary.yaml` which is a good
+starting point for our own pipeline.
+
+Adjust job parameters in `job_train_cls_binary_flowers.yaml`. Many parameters 
+are pre-configured and do not require adjustment. However, we want to configure
+our own 
+- **training** and **prediction** ops (`/ops/train@ops.train.config`, `/ops/train@ops.prediction.config`), 
+- **experiment_id** (`globals` -> `exp_name` and `exp_prefix`)
+- and **input data location** (`globals` -> `data_location`). 
+ 
+In the job configuration, we can already point
+to the location where the Kaggle images are stored. 
+Also, we change the path to the 
+dagster training and prediction operation configurations we are going to write in
+the next two steps.
+
+```yaml
+# train binary classification
+defaults:
+  # experiment
+  - ops/experiment@ops.experiment.config: op_experiment_default.yaml
+  # train
+  - /ops/train@ops.train.config: op_train_cls_binary_flowers.yaml
+  # prediction
+  - /ops/prediction@ops.prediction.config: op_prediction_cls_binary_flowers.yaml
+  # analysis
+  - /ops/analysis@ops.analysis.config: op_analysis_cls_binary.yaml
+  # experiment tests
+  - /ops/exptests@ops.exptests.config.tests: exptests_default.yaml
+  # experiment locations
+  - shared/locations@globals: exp_locations.yaml
+  - _self_
+
+hydra:
+  searchpath:
+    - file://configs
+
+globals:
+  exp_name: FlowersClsBinary
+  exp_prefix: FLOWERSCLB
+  data_location:
+    uri: ${oc.env:DATA_URI,./data}/kaggle_flowers
+```
+
+## Step 3: Configure the training operation
+We navigate to `configs/ops/train` and create `op_train_cls_binary_flowers.yaml`.
```

Comment:
Please replace the ‘we’ statements by simple imperatives to remain consistent with the existing documentation.
This only applies strictly to explicit tasks to do by the user, not to informative/introducing documentation.

Reply:
done.